raven/contrib/celery/__init__.py

```diff
@@ -89,6 +91,16 @@ def process_failure_signal(self, sender, task_id, args, kwargs, einfo, **kw):
 
     def handle_task_prerun(self, sender, task_id, task, **kw):
         self.client.context.activate()
+        if self.context_args:
+            args = inspect.getargspec(task.run)
+            # first arg is going to be self
```

Comment:
I think `arg[0]` might not be `self` if `bind=False` in `@app.task` on the task function. You might want to double check that. I think `task. __bound__` would be `True` when `self` is passed in as the first variable, `False` if not, so you could check that to determine if you need to pop off the first arg or not